tests/lib/newlib/newlib_hook/testcase.yaml

```diff
@@ -0,0 +1,4 @@
+tests:
+  libraries.libc.newlib:
+    tags: clib newlib
+    platform_exclude: native_posix native_posix_64
```

Comment:
Hi, could you please let me know that is there any reason to exclude the native_posix platform?  I did not see the native posix was excluded in another newlib's thread safety testcases.

Reply:
See https://github.com/zephyrproject-rtos/zephyr/pull/38064#discussion_r697794254.